File: 490/Main.java
import java.io.*;

public class Main {
   public static void main(String argv[]) throws IOException {
      int rows, middle, maxLength;
      String[] line = new String[101];
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      rows = maxLength = 0;
      while ((line[rows] = in.readLine()) != null) {
         if (line[rows].length() > maxLength) {
            maxLength = line[rows].length();
            middle = rows;
         }
         ++rows;
      }

      for (int i = 0; i < maxLength; i++) {
         for (int j = rows - 1; j >= 0; j--) {
            if (i < line[j].length()) {
               if (line[j].charAt(i) == '\t')
                  System.out.print(' ');
               else
                  System.out.print(line[j].charAt(i));
            } else {
               System.out.print(' ');
            }
         }
         System.out.println();
      }
   }
}

File: 490/main.c
#include <stdio.h>
#include <memory.h>

int main(int argc, char* argv[])
{
   int i, j;
   int rows, middle, length, max_length = 0;
   char line[100][101];

   memset(line, 0, sizeof(line));

   rows = 0;
   while (gets(line[rows])) {
      length = strlen(line[rows]);
      if (length > max_length) {
         max_length = length;
         middle = rows;
      }
      ++rows;
   }

   for (i = 0; i < max_length; i++) {
      for (j = rows - 1; j >= 0; j--) {
         if (line[j][i] == '\0') {
            if (j > middle) putchar(' ');
         } else {
            if (line[j][i] == '\t')
               putchar(' ');
            else
               putchar(line[j][i]);
         }
      }
      putchar('\n');
   }

   return 0;
}
